### ISSUE TITLE
Add autoscaler - Metrics might not be configured by your cluster administrator - Metrics components already installed .

### ISSUE THREAD

**Issue by Shyckymn** [openshift/origin#10806]
I deployed one mysql image to my openshift infrastructure ("fap" namespace).
When I access the Web Console >"fap" Project > Deployment > and try to "Add autoscaler", I got the message "Metrics might not be configured by your cluster administrator. Metrics are required for autoscaling."
But previously, I installed the Metrics components with success as you can see below.

What else do I have to do? 

```
$ oc login -u system:admin
Logged into "https://10.8.7.213:8443" as "system:admin" using existing credentials.

You have access to the following projects and can switch between them with 'oc project <projectname>':

    default
  * fap
    kube-system
    my-project
    openshift
    openshift-infra

Using project "fap".

$ oc get pods
NAME                     READY     STATUS    RESTARTS   AGE
mysql-55-rhel7-1-dt1uq   1/1       Running   0          16m

$ oc status
In project fap on server https://10.8.7.213:8443

svc/mysql-55-rhel7 - 172.30.116.100:3306
  dc/mysql-55-rhel7 deploys istag/mysql-55-rhel7:latest
    deployment #1 deployed 16 minutes ago - 1 pod

1 warning identified, use 'oc status -v' to see details.


$ oc project openshift-infra
Now using project "openshift-infra" on server "https://10.8.7.213:8443".

$ oc get pods
NAME                         READY     STATUS      RESTARTS   AGE
hawkular-cassandra-1-p766c   1/1       Running     0          3h
hawkular-metrics-pkc0q       1/1       Running     0          3h
heapster-d5yfy               1/1       Running     0          3h
hello-openshift              1/1       Running     0          1h
metrics-deployer-z8cwd       0/1       Completed   0          3h
openshift-router-1-g0edv     1/1       Running     1          3h
```

##### Version
$ oc version
oc v1.3.0-alpha.3+1148c68
kubernetes v1.3.0+507d3a7
features: Basic-Auth GSSAPI Kerberos SPNEGO

Server https://10.8.7.213:8443
openshift v1.3.0-alpha.3+1148c68
kubernetes v1.3.0+507d3a7

**Comment by mwringe** [comment]
Are you able to see metrics in the console? Just wondering if the metrics components are properly running or not.

Can you also try and see if you can access the Heapster endpoint via the proxy? It should be accessible via something like:
`curl -k -H "Authorization: Bearer $CLUSTER_ADMIN_TOKEN" -X GET https://$MASTER_URL:8443/api/v1/proxy/namespace/openshift-infra/services/https:heapster:/api/`

**Comment by Shyckymn** [comment]
@mwringe My pods doesn't have metrics options avaliable in the web console.

$ curl --insecure -H "Authorization: Bearer XXXXXXXXXXXXXXXXX"        -X GET https://10.8.7.213:8443/api/v1/proxy/namespaces/openshift-infra/services/https:heapster:/api/v1/model/metrics
**Unauthorized**

$ oc project openshift-infra
Now using project "openshift-infra" on server "https://10.8.7.213:8443".

$ oc get pods
NAME                         READY     STATUS      RESTARTS   AGE
hawkular-cassandra-1-p766c   1/1       Running     0          22h
hawkular-metrics-pkc0q       1/1       Running     0          22h
heapster-d5yfy               1/1       Running     0          22h
hello-openshift              1/1       Running     0          20h
metrics-deployer-z8cwd       0/1       Completed   0          22h
openshift-router-1-g0edv     1/1       Running     1          23h

**Comment by mwringe** [comment]
@Shyckymn ah, ok, I guess you may not have set the METRICS_URL parameter in the master-config.yaml file. That would explain why the metric tabs are not showing up, this wont affect heapster though.

When you get the unauthorized error trying to access the Heapster api, are you using a token from a cluster-admin or from some other component? I believe its only the cluster admin and hpa sa which can access that endpoint.

If not, did you make any changes to the Heapster configuration when deploying the metric components? Such as changing the secrets for who can access the Heapster api?

**Comment by Shyckymn** [comment]
Hi @mwringe !

I don't make any changes to the Heapster configuration.
I followed the instructions from this link:
https://docs.openshift.org/latest/install_config/cluster_metrics.html

I just forgot to change the master-config.yaml file.
Right now, I have the following configuration:
$ cat /openshift.local.config/master/master-config.yaml
admissionConfig:
(...)
  masterPublicURL: https://10.8.7.213:8443
  metricsPublicURL: "https://10.8.7.213:8443/hawkular/metrics"
  publicURL: https://10.8.7.213:8443/console/

After restart my openshift container, I don't see any changes.
My pod still doesn't have the metrics tab.

$ oc projects
You have access to the following projects and can switch between them with 'oc project <projectname>':

    default
    fap
    kube-system
    my-project - My Project
    openshift
  * openshift-infra

Using project "openshift-infra" on server "https://10.8.7.213:8443".

$ oc status
In project openshift-infra on server https://10.8.7.213:8443

svc/hawkular-cassandra - 172.30.246.186 ports 9042->cql-port, 9160->thift-port, 7000->tcp-port, 7001->ssl-port
  rc/hawkular-cassandra-1 runs openshift/origin-metrics-cassandra:latest
    rc/hawkular-cassandra-1 created 23 hours ago - 1 pod

svc/hawkular-cassandra-nodes (headless) ports 9042, 9160, 7000, 7001
  rc/hawkular-cassandra-1 runs openshift/origin-metrics-cassandra:latest
    rc/hawkular-cassandra-1 created 23 hours ago - 1 pod

https://pxl0docker001 (reencrypt) to pod port https-endpoint (svc/hawkular-metrics)
  rc/hawkular-metrics runs openshift/origin-metrics-hawkular-metrics:latest
    rc/hawkular-metrics created 23 hours ago - 1 pod

svc/heapster - 172.30.239.13:80 -> http-endpoint
  rc/heapster runs openshift/origin-metrics-heapster:latest
    rc/heapster created 23 hours ago - 1 pod

svc/openshift-router - 172.30.67.12 ports 80, 443, 1936
  dc/openshift-router deploys docker.io/openshift/origin-haproxy-router:v1.3.0-alpha.3
    deployment #1 deployed 24 hours ago - 1 pod

pod/hello-openshift runs openshift/hello-openshift

I'm using system:admin account. How can I get it's token to test the Heapster API access directly?

**Comment by Shyckymn** [comment]
$ curl -X GET https://pxl0docker001/hawkular/metrics/status --insecure
{"MetricsService":"STARTED","Implementation-Version":"0.18.3.Final","Built-From-Git-SHA1":"38ea6c2a1e33ed1024684084812ad2ad5730fb76"}

But still without metrics tab...

**Comment by mwringe** [comment]
You should see the metric tabs appear in the console if the metricsPublicURL is set. In your browser, can you directly access `https://10.8.7.213:8443/hawkular/metrics` ? or should it be `https://pxl0docker001/hawkular/metrics` like in your previous curl command?

I have just tried this out, and I can access the Heapster API endpoint using the token from a cluster-admin user:

> curl -k -H "Authorization: Bearer \`oc whoami -t\`" -X GET https://localhost:8443/api/v1/proxy/namespaces/openshift-infra/services/https:heapster:/api/v1/model/metrics

If you cannot access that endpoint using the token from user with cluster-admin privileges, then something is wrong with the policy setup. Does running `oadm policy reconcile-cluster-roles` help?

**Comment by Shyckymn** [comment]
@mwringe, when i try to access https://10.8.7.213:8443/hawkular/metrics, I got the following:

$ oc login -u system:admin
Logged into "https://10.8.7.213:8443" as "system:admin" using existing credentials.

You have access to the following projects and can switch between them with 'oc project <projectname>':

    default
    fap
    kube-system
    my-project
    openshift
  * openshift-infra

Using project "openshift-infra".

Trying to access https://10.8.7.213:8443/hawkular/metrics:

{
  "kind": "Status",
  "apiVersion": "v1",
  "metadata": {},
  "status": "Failure",
  "message": "User \"system:anonymous\" cannot \"get\" on \"/hawkular/metrics\"",
  "reason": "Forbidden",
  "details": {},
  "code": 403

 }

$ oc whoami -t
error: no token is currently in use for this session

**So, I think that the system:admin account cannot get tokens, right?**

The same issue occurs with user "fap":

$ oc login
Authentication required for https://10.8.7.213:8443 (openshift)
Username: fap
Password:
Login successful.

You have one project on this server: "fap"

Using project "fap".

$curl -k -H "Authorization: Bearer `oc whoami -t`" -X GET https://pxl0docker001:8443/api/v1/proxy/namespaces/openshift-infra/services/https:heapster:/api/v1/model/metrics
{
  "kind": "Status",
  "apiVersion": "v1",
  "metadata": {},
  "status": "Failure",
  "message": "User \"fap\" cannot \"proxy\" \"services\" with name \"https:heapster:\" in project \"openshift-infra\"",
  "reason": "Forbidden",
  "details": {
    "name": "https:heapster:",
    "kind": "services"
  },
  "code": 403
}

How can I give cluster-admin roles to my "fap" user?

**Comment by Shyckymn** [comment]
Ooops, I got it:

$ oadm policy add-cluster-role-to-user cluster-admin fap

$ curl -k -H "Authorization: Bearer `oc whoami -t`" -X GET https://pxl0docker001:8443/api/v1/proxy/namespaces/openshift-infra/services/https:heapster:/api/v1/model/metrics
[
  "memory/usage",
  "cpu/request",
  "cpu/limit",
  "memory/request",
  "memory/limit",
  "cpu/usage_rate"

I will restart my openshift container to see what happens now...

**Comment by Shyckymn** [comment]
@mwringe, I'm getting crazy! :-)
After restart, the web console still doesn't show metrics tab in any container...
Is there any command line or logfiles that we should analyse?

$ oc get all
NAME                            REVISION         DESIRED       CURRENT   TRIGGERED BY
dc/openshift-router             1                1             1         config
NAME                            DESIRED          CURRENT       AGE
rc/hawkular-cassandra-1         1                1             1d
rc/hawkular-metrics             1                1             1d
rc/heapster                     1                1             1d
rc/openshift-router-1           1                1             1d
NAME                            HOST/PORT        PATH          SERVICE                               TERMINATION   LABELS
routes/hawkular-metrics         pxl0docker001                  hawkular-metrics:https-endpoint       reencrypt     metrics-infra=hawkular-metrics,name=hawkular-metrics
NAME                            CLUSTER-IP       EXTERNAL-IP   PORT(S)                               AGE
svc/hawkular-cassandra          172.30.246.186   <none>        9042/TCP,9160/TCP,7000/TCP,7001/TCP   1d
svc/hawkular-cassandra-nodes    None             <none>        9042/TCP,9160/TCP,7000/TCP,7001/TCP   1d
svc/hawkular-metrics            172.30.86.49     <none>        443/TCP                               1d
svc/heapster                    172.30.239.13    <none>        80/TCP                                1d
svc/openshift-router            172.30.67.12     <none>        80/TCP,443/TCP,1936/TCP               1d
NAME                            READY            STATUS        RESTARTS                              AGE
po/hawkular-cassandra-1-p766c   1/1              Running       1                                     1d
po/hawkular-metrics-pkc0q       1/1              Running       1                                     1d
po/heapster-d5yfy               1/1              Running       1                                     1d
po/hello-openshift              1/1              Running       0                                     23h
po/metrics-deployer-z8cwd       0/1              Completed     0                                     1d
po/openshift-router-1-g0edv     1/1              Running       1                                     1d

$ curl -k -H "Authorization: Bearer `oc whoami -t`" -X GET https://pxl0docker001:8443/api/v1/proxy/namespaces/openshift-infra/services/https:heapster:/api/v1/model/metrics
[
  "cpu/usage_rate",
  "cpu/request",
  "cpu/limit",
  "memory/request",
  "memory/limit",
  "memory/usage"
 ]

**Comment by mwringe** [comment]
Are you sure you are looking at the right areas in the console?

Can you open the monitoring page and then expand one of the components to see the graphs? https://${MASTER_URL}:8443/console/project/openshift-infra/monitoring

There are few places where the metrics are available in the console, but the monitoring one is probably the quickest one to check.

**Comment by Shyckymn** [comment]
Yeah, I think so...
I'll attach some screen captures...

![open1](https://cloud.githubusercontent.com/assets/21058463/18285121/448ecfb8-7443-11e6-859c-3a148d247dba.JPG)
![open2](https://cloud.githubusercontent.com/assets/21058463/18285120/448aefc4-7443-11e6-829c-360e664e4611.JPG)
![open3](https://cloud.githubusercontent.com/assets/21058463/18285123/44909d66-7443-11e6-96d4-05220b7a30e5.JPG)
![open4](https://cloud.githubusercontent.com/assets/21058463/18285122/448f9be6-7443-11e6-98df-972e2cebe8e6.JPG)
![open5](https://cloud.githubusercontent.com/assets/21058463/18285124/44950f54-7443-11e6-8475-81468bb236e1.JPG)

**Comment by mwringe** [comment]
From the monitoring page, please click the ">" button so that it opens up something beneath that, in that area you should see graphs (or at least a warning message)

From the third screenshot, that one should have the 'metrics' tag available there. The only time I have not seen the metric tab show up there is when the `metricsPublicURL` is not set (or its property value has a spelling mistake in it, such as `MetricsPublicURL` or `metricPublicURL`)

**Comment by Shyckymn** [comment]
Nothing beneath the ">" button.

I haven't found any metrics tag available from the third screenshot...

![open6](https://cloud.githubusercontent.com/assets/21058463/18286142/eea48282-7447-11e6-88be-5dd24ecee6e4.JPG)

**Comment by mwringe** [comment]
Can you please double check the `metricsPublicURL` value in your master-config.yaml? And that you are also picking up that master-config.yaml file and not another one. Please also make sure that you have restarted it after making this change.

**Comment by Shyckymn** [comment]
I've restarted my openshift container and also the k8s containers.
The results from my openshift container:
![open7](https://cloud.githubusercontent.com/assets/21058463/18287484/b250a4a4-744d-11e6-81e2-afadc4ad6231.JPG)

**Comment by mwringe** [comment]
Are you sure that is the master-config.yaml file which is used to start the OpenShift cluster?

Eg it should be started with something like `openshift start --node-config=openshift.local.config/node-mynode/node-config.yaml --master-config=openshift.local.config/master/master-config.yaml`

When you start the OpenShift instance, do you see something like the following in the logs (it should be within the first few messages in the logs):
`start_master.go:272] Warning: assetConfig.metricsPublicURL: Invalid value: "": required to view cluster metrics in the console, master start will continue.`

I cannot reproduce this locally, and as long at the metricsPublicURL is set you should be seeing the metrics tab in the console. If you still cannot get this to function,  we should get this issue assigned to the console team.

**Comment by Shyckymn** [comment]
@mwringe I'm launching the server in a Docker container, using the following command:
```
sudo docker run -d --name "origin" \
        --privileged --pid=host --net=host \
        -v /:/rootfs:ro -v /var/run:/var/run:rw -v /sys:/sys -v /var/lib/docker:/var/lib/docker:rw \
        -v /var/lib/origin/openshift.local.volumes:/var/lib/origin/openshift.local.volumes \
        openshift/origin start
```

My container logfile doesn't have any entry with the "start_master" or "metricsPublicURL" strings.

Do you know if is possible to have the metrics working for a openshift container or should I have to install by "Method 2: Downloading the Binary" from the instructions in https://docs.openshift.org/latest/getting_started/administrators.html?

**Comment by mwringe** [comment]
@Shyckymn running it in an OpenShift container is fine, that should not cause any problems.

@spadgett Any idea why the metrics tab wouldn't be showing up when the metricsPublicURL is set in master-config.yaml?

**Comment by spadgett** [comment]
@Shyckymn A few things to check:

- Is `metricsPublicURL` inside the `assetConfig` stanza in master-config.yaml? It must be under `assetConfig`.
- Do you see any certificate warnings when you open https://10.8.7.213:8443/hawkular/metrics in the browser? If so, you'd need to accept the certificate and reload the web console.

If neither of these is the problems, you should try opening dev tools in Chrome (or Firefox), refreshing the page, and looking at the network tab.

- If you type in "metrics" as a filter, do you see any requests? Are they failing?
- If you type in "config.js" as a filter and click on the request, do you see a `metricsURL` set in the response tab?

https://developer.chrome.com/devtools

**Comment by mwringe** [comment]
Even if the certificate is not accepted, the metric tab for pods should still show up though, right? Its always worked this way for me.

**Comment by spadgett** [comment]
> Even if the certificate is not accepted, the metric tab for pods should still show up though, right? Its always worked this way for me.

In some cases, we stop trying to show metrics if the first request fails (since 1.3). But I checked the code, and the Metrics tab should always show up for pods if the metrics URL is set in master config. I suspect something is wrong in master-config.yaml.

**Comment by Shyckymn** [comment]
@mwringe @spadgett I decided to rebuild my VM and reinstall Openshift, right now, using "Method 2: Downloading the Binary" from the instructions in https://docs.openshift.org/latest/getting_started/administrators.html.
I keep you informed...

**Comment by Shyckymn** [comment]
Steps followed:

`oc login -u system:admin`

`oc project openshift-infra`

```
oc create -f - <<API
apiVersion: v1
kind: ServiceAccount
metadata:
  name: metrics-deployer
secrets:
- name: metrics-deployer
API
```

```
oadm policy add-role-to-user \
    edit system:serviceaccount:openshift-infra:metrics-deployer
```
    
```
oadm policy add-cluster-role-to-user \
    cluster-reader system:serviceaccount:openshift-infra:heapster
```

`oc secrets new metrics-deployer nothing=/dev/null`



`
oc new-app -f metrics-deployer.yaml \
    -p HAWKULAR_METRICS_HOSTNAME=pxl0docker001 \
    -p USE_PERSISTENT_STORAGE=false
`

Results:

oc get pods
NAME                         READY     STATUS    RESTARTS   AGE
hawkular-cassandra-1-4ls61   1/1       Running   0          14m
hawkular-metrics-7xw8y       1/1       Running   0          14m
heapster-cj0qu               1/1       Running   0          14m
metrics-deployer-1o9r6       0/1       Error     0          15m


oc logs metrics-deployer-1o9r6 | tail -10
This often means that the route has already been created (likely in another project) and this one is newer.
It can also mean that no router has been deployed.
---
There was an error while validating the internal hawkular-metrics certificate against the route destination CA:
stdin: CN = hawkular-metrics
error 20 at 0 depth lookup:unable to get local issuer certificate
This will prevent proper functioning of the route.
========================
--- validate_deployed_project ---
VALIDATION FAILED

Any suggests?

**Comment by mwringe** [comment]
@Shyckymn did you install a router in your cluster?

**Comment by Shyckymn** [comment]
@mwringe Yes, I think so....
![open1](https://cloud.githubusercontent.com/assets/21058463/18393421/a3e40a1a-768b-11e6-8771-2a31d820224f.JPG)
![open2](https://cloud.githubusercontent.com/assets/21058463/18393422/a3e41244-768b-11e6-833c-e2741da0f2b3.JPG)

**Comment by mwringe** [comment]
From your screenshot, it looks like the route is setup and running. It might just be an issue with the validation scripts.

What is the output of 'oc get route hawkular-metrics'?

**Comment by Shyckymn** [comment]
Here it goes...
But the strange thing is te WARN message from `oc adm diagnostics`, related to the router configuration...

![open4](https://cloud.githubusercontent.com/assets/21058463/18393729/f700f824-768c-11e6-97ed-e7c7d7b554cf.JPG)

**Comment by Shyckymn** [comment]
@spadgett The `metricsURL `is not set in the response tab:

![open1](https://cloud.githubusercontent.com/assets/21058463/18433067/fab8c4d4-78bb-11e6-9554-6ea8ae4ef094.JPG)

After I reinstalled all the environement (including VM) I searched for a master-config file and it doesn't exists...

![open2](https://cloud.githubusercontent.com/assets/21058463/18433100/2798c666-78bc-11e6-9c27-8cb0f980defa.JPG)

**Comment by spadgett** [comment]
You should see the `--write-config` option to create the file. See the help topic here:

https://docs.openshift.org/latest/install_config/master_node_configuration.html#creating-new-configuration-files

**Comment by Shyckymn** [comment]
@spadgett OK, I run the command `openshift start --write-config=/openshift.local.config`. 
After that, I have the master-config.yaml file in `/openshift.local.config/master/`, different from the path where I'm starting openshift
(`/opt/openshift/openshift-origin-server-v1.3.0-rc1-ac0bb1bf6a629e0c262f04636b8cf2916b16098c-linux-64bit`)

![open0](https://cloud.githubusercontent.com/assets/21058463/18442401/fad35a3a-78e6-11e6-971c-2c454e4bbe38.JPG)


So I edited the master-config.yaml file and added the info related to the metricsURL.

![open3](https://cloud.githubusercontent.com/assets/21058463/18442482/4bc27dfe-78e7-11e6-8770-553e1300de41.JPG)


After that, when I try to start openshift using the command 
`./openshift start master --config=/openshift.local.config/master/master-config.yaml`, 
I got some certificate errors.

![open1](https://cloud.githubusercontent.com/assets/21058463/18442405/fd89c43a-78e6-11e6-9ed9-6fdfee0faf79.JPG)

If I only run `./openshift start` everything works fine...

**Comment by Shyckymn** [comment]
@spadgett @mwringe Any suggestion?

**Comment by spadgett** [comment]
@Shyckymn can you try this command instead, replacing `<node-hostname>` with the correct hostname?

```
$ openshift start --master-config=/openshift.local.config/master/master-config.yaml --node-config=/openshift.local.config/node-<node_hostname>/node-config.yaml
```

**Comment by Shyckymn** [comment]
@spadgett Here are the results... Just remembering the steps:

1. Rebuild my VM and reinstall Openshift, using "Method 2: Downloading the Binary" from the instructions in https://docs.openshift.org/latest/getting_started/administrators.html.

2. Use the `--write-config` option to create the file.

3. So I edited the master-config.yaml file and added the info related to the metricsURL.

![open3](https://cloud.githubusercontent.com/assets/21058463/18513980/08a5377a-7a66-11e6-9921-8443bd17bf86.JPG)

4. Trying to start openshift with command:
`./openshift start --master-config=/openshift.local.config/master/master-config.yaml --node-config=/openshift.local.config/node-pxl0docker001/node-config.yaml

`
![screen1](https://cloud.githubusercontent.com/assets/21058463/18513872/92b06670-7a65-11e6-94bd-e0c40f779930.JPG)

![screen2](https://cloud.githubusercontent.com/assets/21058463/18513873/92b1ba34-7a65-11e6-96da-00deaf391bfd.JPG)

![screen3](https://cloud.githubusercontent.com/assets/21058463/18513874/92b30916-7a65-11e6-9daf-8ba3a9ce4a30.JPG)

If I only run ./openshift start everything works fine...

**Comment by mwringe** [comment]
@Shyckymn You may need to set something like your node's hostname or ip address when calling the write command (eg --hostname ${HOSTNAME}). 

The commands I usually use are (where my local machine hostname is `corbeau`)
```
$ ./openshift start --write-config=openshift.local.config --hostname=corbeau
$ # edit master-config.yaml
$ ./openshift start --node-config=openshift.local.config/node-corbeau/node-config.yaml --master-config=openshift.local.config/master/master-config.yaml
```


In the future can you please attach the files instead of attaching screenshots, its a lot easier to help with issues if we can more easily look through and search things like logs.

@danmcp can you please re-assign this to someone who deals with installation issues? This doesn't appear to be an issue with metrics but with installation.

**Comment by Shyckymn** [comment]
@mwringe @spadgett  I decided to restart the installation/configuration... 

So, I got some news!

Now, I created the master-config.yaml before my first start of openshift. So, the files are under the same tree of my openshift install.

I changed the master-config.yaml contents, to have my metricsPublicURL:
`$ grep metricsPublicURL openshift.local.config/master/master-config.yaml`
`metricsPublicURL: "https://10.8.7.213/hawkular/metrics"`

Started the opensfhit with the command:
`$ ./openshift start --master-config=openshift.local.config/master/master-config.yaml --node-config=openshift.local.config/node-pxl0docker001/node-config.yaml`

Here's the lofgile of my openshift start:

[openshift.txt](https://github.com/openshift/origin/files/472599/openshift.txt)

I also created the router before start metrics install.

Now, when I go to Web Console > Monitoring > and click in a POD, I got some message related to the Metrics:

![screen1](https://cloud.githubusercontent.com/assets/21058463/18519170/435f348e-7a79-11e6-940e-7f1fde18a99d.JPG)

If I try to open the URL, I have the following:

![screen2](https://cloud.githubusercontent.com/assets/21058463/18519196/59d90ed8-7a79-11e6-8126-1c5b6efcfc08.JPG)

In the dev tools from Firefox, I have the following for the metrics call:

![screen10](https://cloud.githubusercontent.com/assets/21058463/18519281/bcd9d274-7a79-11e6-8ab4-e0cbbcc5d9e9.JPG)

And for config.js call:

![screen11](https://cloud.githubusercontent.com/assets/21058463/18519313/e3071966-7a79-11e6-9b76-3ed52ee1661a.JPG)

Excuse me for the screenshots, but in this last post I think that they can be more useful than the logfiles.

**Comment by mwringe** [comment]
Are you sure this is the right IP address for hawkular metrics service?
metricsPublicURL: "https://10.8.7.213/hawkular/metrics"

What is the output for "oc get svc hawkular-metrics"

**Comment by Shyckymn** [comment]
`$ oc get svc hawkular-metrics`
`NAME               CLUSTER-IP     EXTERNAL-IP   PORT(S)   AGE`
`hawkular-metrics   172.30.48.38   <none>        443/TCP   45m`

But this is a internal IP for the container, right? 
I'll not be able to have one answer from the browser to this IP address.
Do I have to change this in the master-config.yaml?

**Comment by mwringe** [comment]
The metricsURL value in master-config.yaml needs to match the following pattern: `https://${HAWKULAR_METRICS_HOSTNAME}/hawkular/metrics` where ${HAWKULAR_METRICS_HOSTNAME} matches the deploymer template parameter.

Using some ip address in the master-config.yaml is not recommended. It can work if you know that is the ip address of the service you want to connect to, but its not a good idea.

You want to use the endpoint exposed by the route, which needs to match your ${HAWKULAR_METRICS_HOSTNAME} value. The router is what exposes your endpoint to outside connections.

Note: setting up and configuring your router, as well as configuring things like how to get the router to work with local DNS servers and such are beyond what will be answered in this issue.

**Comment by Shyckymn** [comment]
@mwringe There are some other entries in the master-config.yaml file which have the server IP address instead the DNS. 
a) Do I have to change them all, or only the line containing assetConfig:metricsPublicURL?
b) Do I have to use port 8443 in the metricsPublicURL?

```
assetConfig:
  extensionDevelopment: false
  extensionProperties: null
  extensionScripts: null
  extensionStylesheets: null
  extensions: null
  loggingPublicURL: ""
  logoutURL: ""
  masterPublicURL: https://10.8.7.213:8443
  metricsPublicURL: "https://pxl0docker001/hawkular/metrics"
  publicURL: https://10.8.7.213:8443/console/
```

```
corsAllowedOrigins:
- 10.8.7.213:8443
- 127.0.0.1
- localhost
```

```
etcdClientInfo:
  ca: ca.crt
  certFile: master.etcd-client.crt
  keyFile: master.etcd-client.key
  urls:
  - https://10.8.7.213:4001
```

```
etcdConfig:
  address: 10.8.7.213:4001
  peerAddress: 10.8.7.213:7001
```

```
kubernetesMasterConfig:
  admissionConfig:
    pluginConfig: null
  apiLevels: null
  apiServerArguments: null
  controllerArguments: null
  disabledAPIGroupVersions: {}
  masterCount: 1
  masterIP: 10.8.7.213
```

`masterPublicURL: https://10.8.7.213:8443`

```
oauthConfig:
  alwaysShowProviderSelection: false
  assetPublicURL: https://10.8.7.213:8443/console/
  masterPublicURL: https://10.8.7.213:8443
  masterURL: https://10.8.7.213:8443
```

**Comment by Shyckymn** [comment]
Hi @mwringe @danmcp @spadgett!
Regarding my last doubt, I only made the changes to the metricsPublicURL parameter. 
The other parameters I left with the IP address of the server.
So, I think that is the right way (Am I right?), because now I have metrics working in my Openshift environment!
I also added my server hostname in my DNS environment.

![open](https://cloud.githubusercontent.com/assets/21058463/18583290/2048dcb2-7be0-11e6-9ed0-b26cac29f46f.JPG)

Thanks for all your attention and support.
